pkg/services/object_manager/tombstone/getter.go

```diff
@@ -0,0 +1,94 @@
+package tombstone
+
+import (
+	"context"
+	"strconv"
+
+	lru "github.com/hashicorp/golang-lru"
+	objectV2 "github.com/nspcc-dev/neofs-api-go/v2/object"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	addressSDK "github.com/nspcc-dev/neofs-sdk-go/object/address"
+	"go.uber.org/zap"
+)
+
+// Source is a tombstone source interface.
+type Source interface {
+	// Tombstone must return tombstone from the source it was
+	// configured to fetch from and any error that appeared during
+	// fetching process.
+	//
+	// Note: it MUST NOT return any "not found"/"deleted" errors.
+	// Missing tombstone must lead to returning nil tombstone pointer
+	// and nil error.
+	Tombstone(ctx context.Context, a *addressSDK.Address, epoch uint64) (*object.Object, error)
+}
+
+// Getter is a tombstone source wrapper.
+// It checks tombstones presence via tombstone
+// source, caches it checks its expiration.
+//
+// Must be created via NewGetter function. `var` and
+// `Getter{}` declarations leads to undefined behaviour
+// and may lead to panics.
+type Getter struct {
```

Comment:
Why is it `Getter` and not `LRUCache` or sth similar?

Reply:
i designed `Getter` as a component that wraps _something_ that could provide info about tombstones. in the first implementation, it contained `local` and `remote` TSSource interfaces: a storage engine and a wrapper over an API client but then it has changed to reusing one object service, and only after that, i added an LRU cache. so I want it to be as abstract as possible

btw, I agree that naming is awful, just could not think of something better, will be happy to hear some suggestions from you

Reply:
`tombstone.HealthChecker`, `tombstone.ExpirationChecker`, `tombstone.Validator`, `tombstone.ExpirationCache`?

Reply:
i have chosen `tombstone.ExpirationChecker`